test/helpers/signPermit.js

```diff
@@ -0,0 +1,59 @@
+const { fromRpcSig } = require('ethereumjs-util')
+
+const EIP712_DOMAIN_TYPE = [
+  { name: 'name', type: 'string' },
+  { name: 'version', type: 'string' },
+  { name: 'chainId', type: 'uint256' },
+  { name: 'verifyingContract', type: 'address' },
+]
+
+const EIP2612_TYPE = [
+  { name: 'owner', type: 'address' },
+  { name: 'spender', type: 'address' },
+  { name: 'value', type: 'uint256' },
+  { name: 'nonce', type: 'uint256' },
+  { name: 'deadline', type: 'uint256' },
+]
+
+module.exports = async ({chainId, account, spender, version, tokenName, tokenAddress, amount, deadline}) => {
+  const message = {
+    owner: account,
+    spender,
+    value: amount,
+    nonce: 0,
+    deadline,
+  }
+  const domain = {
+    name: tokenName,
+    version,
+    verifyingContract: tokenAddress,
+    chainId
+  }
+
+  const dataJSON = {
+    types: {
+      EIP712Domain: EIP712_DOMAIN_TYPE,
+      Permit: EIP2612_TYPE,
+    },
+    domain,
+    primaryType: 'Permit',
+    message
+  }
+  return new Promise((resolve, reject) => {
+    web3.currentProvider.send({
```

Comment:
I believe `web3` functions will default to returning `Promise` if callback is not passed but not 100% sure if that is the case in `currentProvider` as well. We may be using the old pattern in `utils`

Reply:
I must admit I've tried different combinations of functions and async methods and that's the first one that worked after 4 trials. I was looking for `web3.eth.request` (returns a promise and we use it in `dvf-portal`) but couldn't find a direct equivalent for this

Reply:
indeed the signature is painful, I had the hardest time when trying to figure out how to match a sig produced on `solidity` side from `js` side
I can investigate later and see if these have promise equivalent